### PR TITLE
CAD-3444 snocket changes from p2p-master

### PR DESCRIPTION
**Pull request by coot**
- Snocket.Accept
- Use a counter to provide a unique remote addr
- snockets - monadic toBearer
- snocket: support getLocalName for named pipes (Windows)
- snocket: store local and remote path
- snocket: make Window's LocalSocket strict

### CODE REVIEW

**Review by karknu — APPROVED** [review]
LGTM. I think that the last 3 commits could be turned into one commit.

**Comment by coot** [comment]
The first one of these three fixes a snocket api issue: one does not need any more a local address when creating a `Snocket`, the second one make remote address unique, third is an unrelated change which does not fit either purpose.  All in all I prefer to keep them separate.

bors merge

File: ouroboros-network-framework/src/Ouroboros/Network/Snocket.hs [hunk 1]
```diff
@@ -462,7 +468,7 @@ localSnocket ioManager = Snocket {
               -- remote end's address.
               --
               let addr' = LocalAddress $ "\\\\.\\pipe\\ouroboros-network-temp-" ++ show cnt
```
Comment (coot) [inline ouroboros-network-framework/src/Ouroboros/Network/Snocket.hs:470]: better choice: `original@cnt` (for unix & windows)

**Review by coot — COMMENTED** [review]
Some review changes.

File: ouroboros-network-framework/src/Ouroboros/Network/Snocket.hs [hunk 2]
```diff
@@ -125,15 +127,16 @@ instance Bifoldable Accepted where
 berkeleyAccept :: IOManager
                -> Socket
                -> Accept IO Socket SockAddr
-berkeleyAccept ioManager sock = go
+berkeleyAccept ioManager sock = go 0
     where
-      go = Accept (acceptOne `catch` handleException)
+      go cnt = Accept (acceptOne cnt `catch` handleException cnt)
```
[the remark this answers is not in the document]
Comment (coot) [reply ouroboros-network-framework/src/Ouroboros/Network/Snocket.hs:133]: done

File: ouroboros-network-framework/src/Ouroboros/Network/Snocket.hs (hunk 1 above)
Comment (coot) [reply ouroboros-network-framework/src/Ouroboros/Network/Snocket.hs:470]: done

**Pull request merged**